Add cmake FetchContent support

### Suggested enhancement

It would be great if MbedTLS supported cmake's [FetchContent](https://cmake.org/cmake/help/latest/module/FetchContent.html). Ideally a project's CMakeLists.txt file could include:

```
include(FetchContent)

FetchContent_Declare(
        mbedtls
        GIT_REPOSITORY  https://github.com/Mbed-TLS/mbedtls.git
        GIT_TAG         v3.1.0
        GIT_PROGRESS    TRUE
        USES_TERMINAL_DOWNLOAD TRUE)

FetchContent_MakeAvailable(mbedtls)

add_executable(ExampleProgram main.cpp) #includes some MBED code

target_link_libraries(ExampleProgram MbedTLS::mbedcrypto MbedTLS::mbedx509 MbedTLS::mbedtls)
```
### Current behaviour

```
CMake Error at CMakeLists.txt:27 (add_executable):
  Target "ExampleProgram" links to target "MbedTLS::mbedcrypto" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at CMakeLists.txt:27 (add_executable):
  Target "ExampleProgram" links to target "MbedTLS::mbedx509" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?


CMake Error at CMakeLists.txt:27 (add_executable):
  Target "ExampleProgram" links to target "MbedTLS::mbedtls" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
```
### Justification

This would make building from source and linking in client code seamless.

--- a/ChangeLog.d/cmake_add_subdirectory_support.txt
+++ b/ChangeLog.d/cmake_add_subdirectory_support.txt
@@ -1,0 +1,4 @@
+Changes
+   * Add aliases for libraries so that the normal MbedTLS::* targets
+     work when MbedTLS is built as a subdirectory.  Allows use of
+     CMake's FetchContent, as requested in #5688.

--- a/library/CMakeLists.txt
+++ b/library/CMakeLists.txt
@@ -279,6 +279,7 @@ if(USE_SHARED_MBEDTLS_LIBRARY)
 endif(USE_SHARED_MBEDTLS_LIBRARY)
 
 foreach(target IN LISTS target_libraries)
+    add_library(MbedTLS::${target} ALIAS ${target})  # add_subdirectory support
     # Include public header files from /include and other directories
     # declared by /3rdparty/**/CMakeLists.txt. Include private header files
     # from /library and others declared by /3rdparty/**/CMakeLists.txt.

